### DISCUSSION TITLE
Inbox notification filters are completely useless to anyone with a lot of notifications (applies to email as well)

### DISCUSSION THREAD

**Discussion by NSExceptional** [community/community#88749] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

As @chriskuech put it in a similar post, the inbox filtering capabilities don't align at all with how I need to filter my notifications. 99% of the time, what I go to my inbox to do is to see if there are any threads I interacted with that I need to follow up on. To go through the filters one by one:

- **Assigned**: I am never assigned anything directly at work.
- **Participating**: this is where all of my notifications go. I currently have about 620 notifications here.
- **Mentioned**: I am rarely literally @ mentioned, so this is not useful at all really. Would be more useful if it specifically showed comments in threads after I replied in a thread.
- **Team mentioned**: my team is never mentioned either.
- **Review requested**: this is where almost all of my notifications go as well, with about 600 notifications. The discrepancy between this and "participating" can be chalked up to my comments on PRs where my team's review was NOT requested. Somewhat off topic, but I cannot distinguish between MY review being requested, and my TEAM's review being requested.

<img width=30% src=https://github.com/community/community/assets/8371943/e7a62ec5-e625-4d88-9573-113cfe7aa5e8>

Here's an example scenario of my day-to-day workflow:
1. My team's review is requested on a PR (or it is a random PR; makes no difference)
2. I comment on the PR, or review it (note that I was considered "participating" _already_ because my team's review was requested)
3. Hours or days later I realize I want to follow up, so I go to look for it
4. I cannot use the filters to find it because there are dozens of new PRs review requests filling up my inbox, and there is no way to see the things I have explicitly commented on (a PR approval with no remark still counts as a comment; I approve several PRs per day)
5. Sometimes I never find the thing I wanted to follow up on, so I give up; sometimes I get lucky and do find it, usually in my browser history

I also want to be able to identify comments on my PRs without constantly checking them, for all the same reasons above. I cannot distinguish activity on PRs where my review was requested from PRs I opened myself or PRs I directly interacted with. **The way notifications are set up today are completely useless to me;** I have to keep tabs open with the things I know I will want to follow up on. If someone makes a comment on a really old PR or a commit or a gist or something, I will miss it entirely, guaranteed.

I am not the only one with this issue. Here are a handful of similar posts:

- #7850
- #64225
- #44749
- #15817

I have also [voiced my concerns about this before](https://mastodon.social/@objc/111730884681630209) to no avail. This will probably never get any real attention, because none of the other posts did, and none of my cries for help anywhere else do, but I don't know what else to do. If anyone reading this knows how to get in touch with a relevant PM at GitHub, please ping them or let me know!

Pinging @iskae as you commented on a similar post of mine a while back and I'm hoping you're on a relevant team ❤️

**Comment by tinder-cfuller** [comment]
While this may not provide a solution for every one of your use cases, I thought it would not hurt to mention that there are more filters available than the 5 you listed.

https://github.com/notifications?query=reason%3Asecurity-advisory-credit
https://github.com/notifications?query=reason%3Aassign
https://github.com/notifications?query=reason%3Aauthor
https://github.com/notifications?query=reason%3Acomment
https://github.com/notifications?query=reason%3Ainvitation
https://github.com/notifications?query=reason%3Amanual
https://github.com/notifications?query=reason%3Amention
https://github.com/notifications?query=reason%3Areview-requested
https://github.com/notifications?query=reason%3Asecurity-alert
https://github.com/notifications?query=reason%3Astate-change
https://github.com/notifications?query=reason%3Asubscribed
https://github.com/notifications?query=reason%3Ateam-mention
https://github.com/notifications?query=reason%3Aci-activity
https://github.com/notifications?query=reason%3Amember-feature-requested

Any of these can be added to the sidebar for convenience by tapping the gear icon and then `Create new filter`.

<img width="631" alt="Screenshot 2024-01-21 at 3 14 12 PM" src="https://github.com/community/community/assets/22668017/af15705a-2970-4599-b484-3a1c5b8315c2">

And there are [supported is: queries](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/viewing-and-triaging-notifications/managing-notifications-from-your-inbox#supported-is-queries) as well.

**Reply by NSExceptional** [reply]
Hey Chris! I'm familiar with all of those. `reason:commented` is the most useful out of the bunch, but it actually considers an approval ✅ with no actual comment as a "comment", which means PRs I reviewed but didn't leave any feedback on get lumped into that. But imo, the bigger issue with it is…

that it appears to be broken? I only have one result when I use that filter, and I've commented on hundreds of PRs over the years with @tinder-tannerbennett. Maybe it only works on issues for some reason?
<img width="850" alt="Screenshot 2024-01-24 at 3 50 18 PM" src="https://github.com/community/community/assets/8371943/9b610cc5-7313-46ac-98f6-0820bd239675">

**Reply by tinder-cfuller** [reply]
Ah, okay 👍 It may be that, in this context, `comment` might refer specifically to selecting the `comment` option when submitting a code review (as opposed to simply adding an isolated comment). Which might also explain why it's `reason:comment` as opposed to `reason:commented` or something like that. 🤷

**Reply by Iron-Ham** [reply]
As an aside – you may be missing notifications here as you're not SSO'd into either the `TinderBackend` or `TinderEngineering` organizations. As a result, you'll only be shown notifications that don't require SSO to access.

**Reply by NSExceptional** [reply]
@Iron-Ham I an SSO'd into TinderApp which is for our iOS repo, that's where all my ~500 notifications are in the above screenshot. I don't touch any repos in the other two orgs
